gcloud/_helpers.py

```diff
@@ -322,6 +322,49 @@ def _to_bytes(value, encoding='ascii'):
         raise TypeError('%r could not be converted to bytes' % (value,))
 
 
+def _has_field(message_pb, property_name):
+    """Determine if a field is set on a protobuf.
+
+    :type message_pb: :class:`google.protobuf.message.Message`
+    :param message_pb: The message to check for ``property_name``.
+
+    :type property_name: str
+    :param property_name: The property value to check against.
+
+    :rtype: bool
+    :returns: Flag indicating if ``property_name`` is set on ``message_pb``.
+    """
+    # NOTE: As of proto3, HasField() only works for message fields, not for
+    #       singular (non-message) fields. First try to use HasField and
+    #       if it fails (with a ValueError) we manually consult the fields.
```

Comment:
@nathanielmanistaatgoogle Can you chime in here?

What's the correct way to check if a field is set on a `proto3` message?